### ISSUE TITLE
Implement function for model evaluation in forecasting

### ISSUE THREAD

**Issue by mloning** [alan-turing-institute/sktime#648]
**Is your feature request related to a problem? Please describe.**
https://github.com/sktime/enhancement-proposals/pull/8/
#622
#64

**Describe the solution you'd like**
A function to evaluate a forecaster, something along the following lines:

```python
 def evaluate(forecaster, y, fh, X=None, cv=None, strategy="refit", scoring=None):
     """Evaluate forecaster using cross-validation"""
     
     # check cv, compatibility with fh
     # check strategy, e.g. assert strategy in ("refit", "update"), compatibility with cv
     # check scoring
     
     # pre-allocate score array
     n_splits = cv.get_n_splits(y)
     scores = np.empty(n_splits)
     
     for i, (train, test) in enumerate(cv.split(y)):
         # split data
         y_train = y.iloc[train]
         y_test = y.iloc[test]
         # split X too

         # fit and predict
         forecaster.fit(y_train, fh) # pass X too
         y_pred = forecaster.predict()

         # score
         scores[i] = scoring(y_test, y_pred)
     
     # return scores, possibly aggregate
     return scores
 ```

I'll ping you @ViktorKaz as you've worked on this before.

**Comment by thomhomsma** [comment]
My struggle/issue is related to this topic so I will try to give the best possible explanation of what I would like to achieve with Sktime.

The struggle that I am facing is that in the forecasting tutorial of Sktime all examples forecast the full test set.

In my case: I would like to fit the auto ARIMA model on the train set (e.g. 80% of the data). Subsequently, I would like to cross-validate the performance of the model on the test set (final 20% of the data) with a sliding window. Since my forecast horizon is 16 steps, I would like to monitor the performance on the test set for every lead time and the mean performance over all the lead times in a standard error metric (or multiple standard error metrics illustrated in the figure). Since I am comparing the performance to deep learning models that do not use the test data to refit or update their parameters I would only like to fit the ARIMA model on the training data.

I have included two figures to illustrate what my intention is.

![image](https://user-images.githubusercontent.com/73986681/108042640-5ae35d80-7040-11eb-8d0d-45918f737414.png)
![image (1)](https://user-images.githubusercontent.com/73986681/108042691-6cc50080-7040-11eb-966f-91c31b21b6c4.png)

If someone is currently working on this functionality I would be more than happy to help with the development.

**Comment by ViktorKaz** [comment]
was planning to work on this function but I was waiting to see what will be implemented in the [related discussion](https://github.com/alan-turing-institute/sktime/pull/657). I have not checked the full PR request but it looks to me that both the cv object and the evaluate function were implemented by @aiwalter. @thomhomsma does the [PR](https://github.com/alan-turing-institute/sktime/pull/657) address your use case? @mloning I am happy to contribute, le tme know if this issue is still relevant.

**Comment by thomhomsma** [comment]
@ViktorKaz I think to a large extent. However, I would also like a strategy to not update the parameters when checking on the test set.

**Comment by mloning** [comment]
@thomhomsma yes for this we would have to add the `update_params` argument to the `evaluate` function and pass it to the internal `update` function call.

**Comment by mloning** [comment]
Closed by #657
